client/src/components/AddReview/AddReview.js

```diff
@@ -27,6 +27,27 @@ function AddReview() {
   const categories = useSelector(state=>state.categories.allCategories);
 
   const [rateValue, setRateValue] = useState(0);
+  const [reviewExists, setReviewExists] = useState(false);
+  const [existingReview, setExistingReview] = useState({});
+
+  function checkIfReviewExists() {
+    reviews.forEach (
+      (element) => {
+        if (element.user_id === currentUserID && element.place_id === currentPlaceID) {
+          setReviewExists(true);
+          setExistingReview(element);
+          setRateValue(element.rating);
+          return;
+        }
+      }
+    );
+  }
+
+  useEffect(() => {
+    checkIfReviewExists();
```

Comment:
try this and tell me how it goes: 
`  useEffect(checkIfReviewExists, []);`
I really don't like disabling linting.

Reply:
You might need to pass in dependencies or wrap functions in useCallback in order to use them inside useEffect (I ran into this problem when doing the assignments)